internal/docstore/driver/driver.go

```diff
@@ -71,3 +75,28 @@ type Mod struct {
 	FieldPath []string
 	Value     interface{}
 }
+
+// A Query defines a query operation to find documents within a collection based
+// on a set of requirements.
+type Query struct {
+	FieldPaths []string         // the selected field paths
```

Comment:
IMHO these comments need more details.

E.g.,
* for `FieldPath`, add something like: "The returned Documents should only have these fields."
* for `Filters`, add that the filters should be ANDed together
* for `Limit`, can the driver return fewer than Limit even if more are available?
* for `StartAt`, explain where this value will come from; presumably it is from something that the driver previously returned?

Reply:
I doubt we could ever discover the answer to your `Limit` question without reading the server source code. It would be unlikely for that information to be in documentation (for example, the Firestore RPC doc just says that limit is "the maximum number of results to return") and I can't imagine how we could reliably test for it.

Reply:
This is driver documentation; it should specify what acceptable behavior for drivers is, and conformance tests should verify that behavior. This is similar to `PageSize` for blob:
https://github.com/google/go-cloud/blob/master/blob/driver/driver.go#L152

```
// PageSize sets the maximum number of objects to be returned.
// 0 means no maximum; driver implementations should choose a reasonable
// max. It is guaranteed to be >= 0.
```

Reply:
Moved to #1750.